### ISSUE TITLE
Tracking issue for constant prop not causing add'l errors (RFC #1229)

### ISSUE THREAD

**Issue by nikomatsakis** [rust-lang/rust#28238]
Tracking issue for rust-lang/rfcs#1229

cc @oli-obk

**Comment by eddyb** [comment]
Can we move the errors `trans::consts` generates to `check_const`?
E.g. only `bar` causes an error here (a special-case I added before the overflow checks were implemented):
```rust
fn foo<T>() -> u8 { 1 + 255 }
fn bar<T>() -> u8 { 1 / 0 }
```
`foo` only errors if it is monomorphized, even though the overflow exists in the generic definition.

**Comment by oli-obk** [comment]
@eddyb: that sounds reasonable as a second step, but is orthogonal to the RFC. I also don't want to create a humongous PR, so I'll address the port to `check_const` right afterwards. It's probably better to create a new issue (and assign it to me if you want)

**Comment by brson** [comment]
@eddyb This is ancient. Can it move forward yet?

**Comment by eddyb** [comment]
@brson Nope, this is blocked on miri!

**Comment by Aaronepower** [comment]
@eddyb What is currently blocking this issue?

**Comment by oli-obk** [comment]
The new const propagator needs to be fixed to actually lint these functions

1. even if they are unused (make them nongeneric and don't use them -> no lint)
2. even if they are generic (currently generic functions seem to not get linted)

**Comment by joshtriplett** [comment]
ping @RalfJung @oli-obk
Is this currently done? Based on the description, this *sounds* like something that's working now.

**Comment by RalfJung** [comment]
Yeah I think this is implemented, but I also don't know what @oli-obk had in mind in their last comment here. @oli-obk what would be an example for something that fails (failed?) to get the lint?

**Comment by oli-obk** [comment]
We are now checking everything properly. I don't remember if my comment was wrong or if it got addressed.
